File: sdkwork-ai-sdk-python/sdkwork_ai_sdk/api/embedding.py
from typing import Any, Dict, List, Optional
from ..http_client import HttpClient
from ..models import EmbeddingRequest, EmbeddingResponse

class EmbeddingApi:
    """embedding API client."""
    
    def __init__(self, client: HttpClient):
        self._client = client

    def create_embedding(self, body: EmbeddingRequest) -> EmbeddingResponse:
        """Create embeddings"""
        return self._client.post(f"/ai/v3/embeddings", json=body)

File: sdkwork-ai-sdk-flutter/lib/src/api/embedding.dart
import '../http/client.dart';
import '../models.dart';

class EmbeddingApi {
  final HttpClient _client;
  
  EmbeddingApi(this._client);

  /// Create embeddings
  Future<EmbeddingResponse?> createEmbedding(EmbeddingRequest body) async {
    final response = await _client.post(ApiPaths.aiPath('/embeddings'), body: body, contentType: 'application/json');
    return response is EmbeddingResponse ? response : null;
  }
}
